python/triton/language/semantic.py

```diff
@@ -1347,6 +1348,24 @@ def assert_dtypes_valid(lhs_dtype, rhs_dtype, options):
         _0 = builder.get_fp16(0) if out_dtype.is_fp16() else builder.get_fp32(0)
         ret_scalar_ty = out_dtype
 
+    if not allow_tf32 and lhs.dtype.is_fp32() and rhs.dtype.is_fp32() and getattr(builder.options, "supports_tf32"):
```

Comment:
What if the builder doesn't have a `supports_tf32` option?

Reply:
getattr will return None and it'll evaluate to False.